internal/common/mqtt/mqttconfig.go

```diff
@@ -92,21 +107,36 @@ func addClientData(client *Client, clientID string) {
 }
 
 //SetBrokerURL returns the broker url for connection
-func (c *Client) SetBrokerURL(protocol string) string {
-	return fmt.Sprintf("%s://%s:%d", protocol, c.Host, c.Port)
+func (c *Client) SetBrokerURL() string {
+	return fmt.Sprintf("%s://%s:%d", c.protocol, c.Host, c.Port)
 }
 
-func checkforConnection(brokerURL string, mqttClient *Client) int {
+func checkforConnection(brokerURL string, mqttClient *Client, mqttPort uint) int {
 	if mqttClient == nil {
 		return 0
 	}
 	log.Info(logPrefix, mqttClient.URL)
-	connURL := fmt.Sprintf("%s://%s:%d", "tcp", brokerURL, mqttPort)
+	connURL := fmt.Sprintf("%s://%s:%d", mqttClient.protocol, brokerURL, mqttPort)
 	return strings.Compare(connURL, mqttClient.URL)
 }
 
+//NewTLSConfig creates a tls config for mqtt client
+func NewTLSConfig() *tls.Config {
+	certpool := x509.NewCertPool()
+	ca, err := ioutil.ReadFile(caCertConfig)
+	if err != nil {
+		log.Warn(err.Error())
+		return nil
+	}
+	certpool.AppendCertsFromPEM(ca)
+	return &tls.Config{
+		RootCAs: certpool,
+	}
+}
+
 // NewClient returns a configured `Client`. Is mandatory
 func NewClient(configs ...Config) (*Client, error) {
+	tlsconfig := NewTLSConfig()
```

Comment:
If `NewTLSConfig` returns an error, `NewClient` also should returns an error.

Reply:
Done!!